src/rp2_common/pico_stdlib/stdlib.c

```diff
@@ -36,6 +36,16 @@ void set_sys_clock_48mhz() {
     }
 }
 
+// When the System clock PLL setup is changed from the initial setting made
+// during system initialisation, the code automatically switches the peripheral
+// clock to the 48MHz USB clock to ensure continuity of peripheral operation.
+// The Pico config option below allows a user to change this default behaviour if
+// required by defining the Pico config option below as 0 in their board file.
+// PICO_CONFIG: PICO_CLOCK_AJDUST_PERI_CLOCK_WITH_SYS_CLOCK, When the SYS clock PLL is changed also switch the peripheral clock to the USB clock, type=bool, default=1, advanced=true, group=hardware_clocks
+#ifndef PICO_CLOCK_AJDUST_PERI_CLOCK_WITH_SYS_CLOCK
```

Comment:
Not sure if this is the best name? "AJDUST_PERI_CLOCK_WITH_SYS_CLOCK" _might_ be misinterpreted to mean that the peri-clock will be adjusted to the same value as the sys-clock? From my limited understanding of the PLL and Clocks code, this setting is effectively `PICO_CLOCK_DETACH_PERI_CLOCK_FROM_SYS_CLOCK_WHEN_CHANGING_SYS_CLOCK` ? :wink: 

Perhaps it's worth inverting the logic, and instead adding `PICO_CLOCK_KEEP_PERI_CLOCK_ATTACHED_TO_SYS_CLOCK` which defaults to `0`? :man_shrugging:

Reply:
Yes the default should be zero.
Other note, you should probably move the define to the stdlib.h header (so that other code can see the value)

Reply:
At this point I can see value in lurch's suggestion:
`PICO_CLOCK_KEEP_PERI_CLOCK_ATTACHED_TO_SYS_CLOCK`
I'd like to use that please.